packages/web3-eth-accounts/src/index.js

```diff
@@ -264,18 +271,20 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
 
     // Otherwise, get the missing info from the Ethereum Node
     return Promise.all([
-        isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId,
+        ((isNot(tx.common) || isNot(tx.common.customChain.chainId)) ? //tx.common.customChain.chainId is not optional inside tx.common if tx.common is provided
+            ( isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId) 
+            : undefined ), 
         isNot(tx.nonce) ? _this._ethereumCall.getTransactionCount(_this.privateKeyToAccount(privateKey).address) : tx.nonce,
         isNot(hasTxSigningOptions) ? _this._ethereumCall.getNetworkId() : 1,
         _handleTxPricing(_this, tx)
     ]).then(function(args) {
-        if (isNot(args[0]) || isNot(args[1]) || isNot(args[2]) || isNot(args[3])) {
+        if ( (isNot(args[0]) && isNot(tx.common) && isNot(tx.common.customChain.chainId)) || isNot(args[1]) || isNot(args[2]) || isNot(args[3])) {
```

Comment:
Please use the destruct for the args array to improve the code readability. 

```javascript
const [chainId, nonce, networkId, gasInformation] = args; 
```

Reply:
yes I was aware of it already and there are 100s of places in 1.x for **readability** improvement, that why we will only focus our energies on only functional changes in 1.x and will put energy in 4.x

Reply:
Yes I believe there could be more 100 more places, but the code base we touch upon must be touch up a little. Refactoring code is on-going process, there can't be a time where we say let's refactor all complex conditional checks in the library, rather we keep improving which ever code we touch.